rasa/core/test.py

```diff
@@ -539,7 +539,15 @@ async def _predict_tracker_actions(
             num_predicted_actions += 1
 
         elif use_e2e and isinstance(event, UserUttered):
-            predicted = await processor.parse_message(UserMessage(event.text))
+            # This means that user utterance didn't have a user message, only intent,
+            # so we can skip the NLU part and take the parse data directly.
+            # Indirectly that means that the test story was in YAML format.
+            if not event.text:
```

Comment:
I think we should print a warning in this case. Doesn't make sense to have labeled data without text in the tests. Might be even worth to throw an exception at the user

Reply:
Imagine a situation when there was a button with intents - there's no user utterance, only intent.
These stories are still valid and should be testable.

Reply:
still. Then they should have text `/greet` or so.

Reply:
I think more feedback for the user would be preferable, but up to you.

Reply:
Hm, not throwing an exception is the whole point here, I wanted to make this test story:
```yaml
stories:
- story: happy path 1
  steps:
  - intent: greet
  - action: utter_greet
  - intent: mood_great
  - action: utter_happy
```
.. a valid one. Otherwise they will look like this:
```yaml
stories:
- story: happy path 1
  steps:
  - intent: greet
     user: /greet
  - action: utter_greet
  - intent: mood_great
     user: /mood_great
  - action: utter_happy

```
.. which takes away advantages of using explicit `intent` key.

Reply:
@akelad wdyt?